bulk_indexer.go

```diff
@@ -351,10 +351,19 @@ func (b *BulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	b.bytesUncompFlushed = bytesUncompFlushed
 	var resp BulkIndexerResponseStat
 	if res.IsError() {
-		if res.StatusCode == http.StatusTooManyRequests {
-			return resp, errorTooManyRequests{res: res}
+		e := errorFlushFailed{resp: res.String(), statusCode: res.StatusCode}
+		if res.StatusCode >= 400 {
```

Comment:
go-elasticsearch client treats `StatusCode > 299` as Error [here](https://github.com/elastic/go-elasticsearch/blob/v8.14.0/esapi/esapi.response.go#L92). I'm not sure how relevant handling of 3xx is, but right now any 3xx will be skipped in [here](https://github.com/elastic/go-docappender/pull/177/files#diff-e8795b30d8c0f7710cb5b474d8333d8ffacaadc4a577bbae4a8c1c2a732f4f30R362)